### ISSUE TITLE
Missing functions: `d_strcmp*`, `d_pavcmp`

### ISSUE THREAD

**Issue by andrewseidl** [gpuopenanalytics/CSVImporter#1]
I'm receiving quite a few undefined identifiers in `deviceutils.cuh`, saying that `d_strcmp` and related functions don't exist. For example, at: https://github.com/gpuopenanalytics/CSVImporter/blob/baae6e9bcc5dad12c11bae6a16033f8726365b0c/include/device/devicetypes.cuh#L69

It doesn't seem like these specific functions are from the original ModernGPU repo, so I'm wondering if a file might be missing?

```
device/devicetypes.cuh(69): error: identifier "d_strcmp" is undefined
device/devicetypes.cuh(75): error: identifier "d_strcmp" is undefined
device/devicetypes.cuh(82): error: identifier "d_strcmpi" is undefined
device/devicetypes.cuh(88): error: identifier "d_strcmpi" is undefined
device/devicetypes.cuh(95): error: identifier "d_strcmp8" is undefined
device/devicetypes.cuh(101): error: identifier "d_strcmp8" is undefined
device/devicetypes.cuh(107): error: identifier "d_strcmpi8" is undefined
device/devicetypes.cuh(113): error: identifier "d_strcmpi8" is undefined
device/devicetypes.cuh(120): error: identifier "d_pavcmp" is undefined
device/devicetypes.cuh(126): error: identifier "d_pavcmp" is undefined
```

I've started adding Linux support (https://github.com/andrewseidl/CSVImporter/tree/linux-support), so it's entirely possible that I missed an include or library somewhere (`CMakeLists.txt` needs a bit of a rearrange based on how things are setup in VS; I simply tossed everything into a single lib). However it didn't look like there were any non-standard, non-CUDA libraries being pulled in, and a quick grep of the CUDA toolkit didn't reveal any of the missing functions.

**Comment by ghost** [comment]
Looking into it...

**Comment by ghost** [comment]
We tried to bring over as little as possible from ModernGPU, and we enhanced some features.  Those set of functions (d_strcmp ... d_pavcmp .. d_skucmp) are not used by this project (leftover defines from another project) and can be removed from the devicetypes.cuh file (or commented out).  Apparently Linux compilers error if a function prototype doesn't have a corresponding function definition in the project (where as Visual Studio just ignores it - so long as its not referenced anywhere).

I have updated the devicetypes.cuh on the Simantex GitHub, so GOAi could update its Fork if desired.
I have rebuilt and tested - all still works fine without those functions.

**Comment by andrewseidl** [comment]
That fixed it, thanks!

Resolved by 9787b97f3e88c1b488c1bc0e1b7bf880e3fc1618
